CMakeLists.txt

```diff
@@ -2,29 +2,87 @@
 
 cmake_minimum_required(VERSION 3.25)
 
-project(beman.scope DESCRIPTION "Generic Scope Guard" LANGUAGES CXX)
-
-enable_testing()
+project(
+    beman.scope
+    DESCRIPTION "Generic Scope Guard"
+    LANGUAGES CXX
+    VERSION 0.0.1
+)
 
 # [CMAKE.SKIP_TESTS]
 option(
     BEMAN_SCOPE_BUILD_TESTS
-    "Enable building tests and test infrastructure. Default: ON. Values: { ON, OFF }."
+    "Enable building tests and test infrastructure. Default: ${PROJECT_IS_TOP_LEVEL}. Values: { ON, OFF }."
     ${PROJECT_IS_TOP_LEVEL}
 )
 
 # [CMAKE.SKIP_EXAMPLES]
 option(
     BEMAN_SCOPE_BUILD_EXAMPLES
-    "Enable building examples. Default: ON. Values: { ON, OFF }."
+    "Enable building examples. Default: ${PROJECT_IS_TOP_LEVEL}. Values: { ON, OFF }."
+    ${PROJECT_IS_TOP_LEVEL}
+)
+
+option(
+    BEMAN_SCOPE_INSTALL_CONFIG_FILE_PACKAGE
+    "Enable creating and installing a CMake config-file package. Default: ${PROJECT_IS_TOP_LEVEL}. Values: { ON, OFF }."
     ${PROJECT_IS_TOP_LEVEL}
 )
 
-include(GNUInstallDirs)
+add_library(beman.scope INTERFACE)
+add_library(beman::scope ALIAS beman.scope)
+
+# gersemi: off
+set_target_properties(
+    beman.scope
+    PROPERTIES
+        VERIFY_INTERFACE_HEADER_SETS ON
+        EXPORT_NAME scope
+)
+
+target_sources(
+    beman.scope
+    INTERFACE
+        FILE_SET HEADERS
+        BASE_DIRS include
+        FILES include/beman/scope/scope.hpp
+)
+
+install(
+    TARGETS beman.scope
+    EXPORT beman.scope-targets
+    COMPONENT beman.scope
+    FILE_SET HEADERS
+)
+# gersemi: on
+
+if(BEMAN_SCOPE_INSTALL_CONFIG_FILE_PACKAGE)
+    include(GNUInstallDirs)
+    include(CMakePackageConfigHelpers)
+
+    write_basic_package_version_file(
+        ${CMAKE_CURRENT_BINARY_DIR}/beman.scope-config-version.cmake
+        COMPATIBILITY ExactVersion
+    )
 
-# todo rm add_subdirectory(src/beman/scope)
+    install(
+        FILES
+            cmake/beman.scope-config.cmake
+            ${CMAKE_CURRENT_BINARY_DIR}/beman.scope-config-version.cmake
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/beman.scope
+        COMPONENT beman.scope
+    )
+
+    install(
+        EXPORT beman.scope-targets
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/beman.scope
+        NAMESPACE beman::
+        COMPONENT beman.scope
+    )
+endif()
 
 if(BEMAN_SCOPE_BUILD_TESTS)
+    enable_testing()
```

Comment:
another case where exemplar perhaps should be updated -- in that repo ctest is unconditionally included in the top level project

Reply:
This has been updated multiple time, the consensus formed is to just use `include(CTest)`.

See: https://github.com/bemanproject/exemplar/pull/153

Reply:
I don't think https://github.com/bemanproject/exemplar/pull/153 or https://github.com/bemanproject/exemplar/pull/109 mentioned anything about dashboard.

Reply:
It's more about consistently define the test target.

Please check the back-and-forth reasoning behind these PRs:
https://github.com/bemanproject/exemplar/pull/153
https://github.com/bemanproject/exemplar/pull/109
https://github.com/bemanproject/exemplar/pull/91

Reply:
Didn't realize this was a beehive already thrice stepped on

Reply:
There's a couple lines in exemplar's CMake that has undergone a lot of discussion and back-and-forth edits.
It's not like we can put a #WARNING: BEFORE YOU UPDATE, CHECK [PR numbers....]

Reply:
I think the expectation is that we add something to the beman standard explaining when we have a recommendation or a requirement. I won't have time in the next few weeks to follow up on that, so if someone else wants to take a pass, please do. Otherwise, hopefully it can wait my other activities (like finalizing the CppCon program) can wrap up.